modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CSharpClientCodegen.java

```diff
@@ -1570,38 +1565,24 @@ public ModelsMap postProcessModels(ModelsMap objs) {
 
     // https://github.com/OpenAPITools/openapi-generator/issues/15867
     @Override
-    protected void removePropertiesDeclaredInComposedTypes(Map<String, ModelsMap> objs, CodegenModel model, List<CodegenProperty> composedProperties) {
+    protected void removePropertiesAlreadyDeclaredInSubModel(Map<String, ModelsMap> models, CodegenModel superModel) {
         if (!GENERICHOST.equals(getLibrary())) {
             return;
         }
 
-        String discriminatorName = model.discriminator == null
-                ? null
-                : model.discriminator.getPropertyName();
-
-        for (CodegenProperty oneOfProperty : composedProperties) {
-            String ref = oneOfProperty.getRef();
-            if (ref != null) {
-                for (Map.Entry<String, ModelsMap> composedEntry : objs.entrySet()) {
-                    CodegenModel composedModel = ModelUtils.getModelByName(composedEntry.getKey(), objs);
-                    if (ref.endsWith("/" + composedModel.name)) {
-                        for (CodegenProperty composedProperty : composedModel.allVars) {
-                            if (discriminatorName != null && composedProperty.name.equals(discriminatorName)) {
-                                continue;
```

Comment:
The majority of the fix is in removing this if-statement, so that discriminators are also removed from super classes.